### PR TITLE
Fix new home additions

### PR DESCRIPTION
**Pull request by drewvolz**
The home screen became editable as of a few PRs ago! You can reorder the views to your liking, or to your friend's disliking!

So what happens when we add a new view to the home screen? It doesn't show in the list of views you can reorder. Oh no... how do we fix this? Well, we could check if we have a differing amount of views between what we _know_ is your preferred order, and the list of views we know are displaying by default.

That seems to work pretty well! We can now get away with safely adding or removing views from the home screen without any issues. This is good news, because we want the user to be able to reorder all of the views, not just some of them.

This was a combo effort by @hawkrives and myself. We even have tests to prove it works (thank you hawken).

tl;dr: adding/removing views did not work with the edit stuff, and this fixes that.

### CODE REVIEW

**Comment by hawkrives** [comment]
This will also do its best to maintain the sort order that you had before we changed things around.

It will remove views from their spots in the sorted list, and add any new ones to the end.

`saved: [a b c]` `new: [a c d]` -> `result: [a c d]`

**Review by hawkrives — APPROVED** [review]
LGTM
